### PR TITLE
Add Qk100 Keyboard files

### PR DESCRIPTION
**Pull request by owlab-git**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/qwertykeys/qk100/ansi/ansi.c [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
+
```
Comment (waffle87) [inline keyboards/qwertykeys/qk100/ansi/ansi.c:219]: Handled by `RGB_DISABLE_WHEN_USB_SUSPENDED`.
```suggestion
```

File: keyboards/qwertykeys/qk100/ansi/ansi.c [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+void keyboard_post_init_kb(void) {
+    rgblight_layers = my_rgb_layers;
+    keyboard_post_init_user();
+}
+
+bool led_update_kb(led_t led_state) {  
+    rgblight_set_layer_state(0, led_state.caps_lock);
+    rgblight_set_layer_state(1, !led_state.num_lock);
+    return false;
```
Comment (waffle87) [inline keyboards/qwertykeys/qk100/ansi/ansi.c:247]:
```suggestion
    bool res = led_update_user(led_state);
    if (res) {
        rgblight_set_layer_state(0, led_state.caps_lock);
        rgblight_set_layer_state(1, !led_state.num_lock);
    }
    return res; 
```

File: keyboards/qwertykeys/qk100/ansi/ansi.c [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "ansi.h"
```
Comment (waffle87) [inline keyboards/qwertykeys/qk100/ansi/ansi.c:17]:
```suggestion
#include "quantum.h"
```

File: keyboards/qwertykeys/qk100/ansi/ansi.h
Comment (waffle87) [inline keyboards/qwertykeys/qk100/ansi/ansi.h:1]: This file should be removed.

File: keyboards/qwertykeys/qk100/ansi/config.h [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+#    define RGBLIGHT_EFFECT_BREATHE_CENTER 1.0
+#    define RGBLIGHT_EFFECT_BREATHE_MAX    60
+
+
+/* RGB matrix */
+#ifdef RGB_MATRIX_ENABLE
+// i2c setting
+#define USE_I2CV1
+#define I2C1_SCL 8  
+#define I2C1_SDA 9
```
Comment (waffle87) [inline keyboards/qwertykeys/qk100/ansi/config.h:39]: `USE_I2CV1` is conditionally defined for this MCU, and `I2C1_SDA/SCL` defines do not exist.
```suggestion
```

File: keyboards/qwertykeys/qk100/ansi/config.h [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+/* RGB matrix */
+#ifdef RGB_MATRIX_ENABLE
+// i2c setting
+#define USE_I2CV1
+#define I2C1_SCL 8  
+#define I2C1_SDA 9  
+#define I2C1_CLOCK_SPEED 400000  
+#define I2C1_DUTY_CYCLE FAST_DUTY_CYCLE_2  
+#    define RGB_MATRIX_DISABLE_KEYCODES 
+#    define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
```
Comment (waffle87) [inline keyboards/qwertykeys/qk100/ansi/config.h:43]:
```suggestion
```

File: keyboards/qwertykeys/qk100/ansi/halconf.h [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define HAL_USE_I2C TRUE
+
+#define HAL_USE_SPI TRUE
+
```
Comment (waffle87) [inline keyboards/qwertykeys/qk100/ansi/halconf.h:22]: This doesn't look used.
```suggestion
```

File: keyboards/qwertykeys/qk100/solder/halconf.h
Comment (waffle87) [inline keyboards/qwertykeys/qk100/solder/halconf.h:1]: These settings don't seem to apply anywhere for this board. This file should be removed.

File: keyboards/qwertykeys/qk100/solder/mcuconf.h
Comment (waffle87) [inline keyboards/qwertykeys/qk100/solder/mcuconf.h:1]: These settings don't seem to apply anywhere for this board. This file should be removed.

File: keyboards/qwertykeys/qk100/solder/solder.h
Comment (waffle87) [inline keyboards/qwertykeys/qk100/solder/solder.h:1]: This file should be removed.

File: keyboards/qwertykeys/qk100/solder/solder.c [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "solder.h"
```
Comment (waffle87) [inline keyboards/qwertykeys/qk100/solder/solder.c:17]:
```suggestion
#include "quantum.h"
```

File: keyboards/qwertykeys/qk100/solder/solder.c [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+void keyboard_post_init_kb(void) {
+    rgblight_layers = my_rgb_layers;
+    keyboard_post_init_user();
+}
+
+bool led_update_kb(led_t led_state) {  
+    rgblight_set_layer_state(0, led_state.caps_lock);
+    rgblight_set_layer_state(1, !led_state.num_lock);
+    return false;
```
Comment (waffle87) [inline keyboards/qwertykeys/qk100/solder/solder.c:44]:
```suggestion
    bool res = led_update_user(led_state);
    if (res) {
        rgblight_set_layer_state(0, led_state.caps_lock);
        rgblight_set_layer_state(1, !led_state.num_lock);
    }
    return res; 
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by waffle87** [comment]
It would also be preferable to condense the duplicated code, such as
```shell
qk100
├── ansi
│   └── info.json # ansi specific configuration
├── info.json # configuration that applies to both pcbs
├── ld
├── qk100.c
├── rules.mk
└── solder
    └── info.json # solder specific configuration
```

File: keyboards/qwertykeys/qk100/ansi/config.h (hunk 5 above)
Comment (owlab-git) [reply keyboards/qwertykeys/qk100/ansi/config.h:43]: Hi, this section is to disable rgb matrix reacting to RGB KEYCODE but the f14-22, else RGB KEYCODE will work on rgblight and rgb matrix at the sametime, which is not our intended behavior. Thanks anyway.

**Comment by owlab-git** [comment]
Bump

File: keyboards/qwertykeys/qk100/ansi/ansi.c [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+    { 0, 64 }, { 14, 64 }, { 28, 64 }, { 42, 64 },             { 74, 64 },                          { 108, 64 }, { 122, 64 }, { 140, 64 },             { 158, 64 }, { 170, 64 }, { 182, 64 }, { 196, 64 },              { 212, 64 }
+}, {
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,       1, 1, 1, 1,
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1,    1, 1, 1, 1,
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1, 4, 4, 4,
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,    1, 1, 4, 4, 4, 1,
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1,    1,    4, 4, 4, 1,
+    1, 1, 1, 4,    4,       4, 1, 1,    1, 1, 1, 4,    4
+    }
+};
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/ansi/ansi.c:151]: This can and should be migrated to info.json.  
```suggestion
```

File: keyboards/qwertykeys/qk100/ansi/info.json [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+        "pid": "0x3030",
+        "device_version": "0.0.1"
+    },
+    "features": {
+        "rgb_matrix": true
+    },
+    "rgb_matrix": {
+        "driver": "IS31FL3741",
+        "max_brightness": 200,
+        "animations": {
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/ansi/info.json:13]: led config should look like this, but please verify.
```suggestion
        "layout": [
            {"matrix": [0, 0], "x": 0, "y": 0, "flags": 1},
            {"matrix": [0, 1], "x": 18, "y": 0, "flags": 1},
            {"matrix": [0, 2], "x": 30, "y": 0, "flags": 1},
            {"matrix": [0, 3], "x": 42, "y": 0, "flags": 1},
            {"matrix": [0, 4], "x": 54, "y": 0, "flags": 1},
            {"matrix": [0, 5], "x": 72, "y": 0, "flags": 1},
            {"matrix": [0, 6], "x": 84, "y": 0, "flags": 1},
            {"matrix": [0, 7], "x": 96, "y": 0, "flags": 1},
            {"matrix": [0, 8], "x": 108, "y": 0, "flags": 1},
            {"matrix": [0, 9], "x": 126, "y": 0, "flags": 1},
            {"matrix": [0, 10], "x": 138, "y": 0, "flags": 1},
            {"matrix": [0, 11], "x": 150, "y": 0, "flags": 1},
            {"matrix": [0, 12], "x": 162, "y": 0, "flags": 1},
            {"matrix": [0, 15], "x": 188, "y": 0, "flags": 1},
            {"matrix": [0, 16], "x": 200, "y": 0, "flags": 1},
            {"matrix": [0, 17], "x": 212, "y": 0, "flags": 1},
            {"matrix": [0, 18], "x": 224, "y": 0, "flags": 1},
            {"matrix": [1, 0], "x": 0, "y": 13, "flags": 4},
            {"matrix": [1, 1], "x": 12, "y": 13, "flags": 4},
            {"matrix": [1, 2], "x": 24, "y": 13, "flags": 4},
            {"matrix": [1, 3], "x": 36, "y": 13, "flags": 4},
            {"matrix": [1, 4], "x": 48, "y": 13, "flags": 4},
            {"matrix": [1, 5], "x": 60, "y": 13, "flags": 4},
            {"matrix": [1, 6], "x": 72, "y": 13, "flags": 4},
            {"matrix": [1, 7], "x": 84, "y": 13, "flags": 4},
            {"matrix": [1, 8], "x": 96, "y": 13, "flags": 4},
            {"matrix": [1, 9], "x": 108, "y": 13, "flags": 4},
            {"matrix": [1, 10], "x": 120, "y": 13, "flags": 4},
            {"matrix": [1, 11], "x": 132, "y": 13, "flags": 4},
            {"matrix": [1, 12], "x": 144, "y": 13, "flags": 4},
            {"matrix": [1, 13], "x": 156, "y": 13, "flags": 1},
            {"matrix": [1, 15], "x": 188, "y": 13, "flags": 1},
            {"matrix": [1, 16], "x": 200, "y": 13, "flags": 1},
            {"matrix": [1, 17], "x": 212, "y": 13, "flags": 1},
            {"matrix": [1, 18], "x": 224, "y": 13, "flags": 1},
            {"matrix": [2, 0], "x": 0, "y": 26, "flags": 1},
            {"matrix": [2, 1], "x": 18, "y": 26, "flags": 4},
            {"matrix": [2, 2], "x": 30, "y": 26, "flags": 4},
            {"matrix": [2, 3], "x": 42, "y": 26, "flags": 4},
            {"matrix": [2, 4], "x": 54, "y": 26, "flags": 4},
            {"matrix": [2, 5], "x": 66, "y": 26, "flags": 4},
            {"matrix": [2, 6], "x": 78, "y": 26, "flags": 4},
            {"matrix": [2, 7], "x": 90, "y": 26, "flags": 4},
            {"matrix": [2, 8], "x": 102, "y": 26, "flags": 4},
            {"matrix": [2, 9], "x": 114, "y": 26, "flags": 4},
            {"matrix": [2, 10], "x": 126, "y": 26, "flags": 4},
            {"matrix": [2, 11], "x": 138, "y": 26, "flags": 4},
            {"matrix": [2, 12], "x": 150, "y": 26, "flags": 4},
            {"matrix": [2, 13], "x": 162, "y": 26, "flags": 4},
            {"matrix": [2, 14], "x": 176, "y": 26, "flags": 1},
            {"matrix": [2, 15], "x": 188, "y": 26, "flags": 4},
            {"matrix": [2, 16], "x": 200, "y": 26, "flags": 4},
            {"matrix": [2, 17], "x": 212, "y": 26, "flags": 4},
            {"matrix": [3, 0], "x": 0, "y": 39, "flags": 1},
            {"matrix": [3, 1], "x": 20, "y": 39, "flags": 4},
            {"matrix": [3, 2], "x": 32, "y": 39, "flags": 4},
            {"matrix": [3, 3], "x": 44, "y": 39, "flags": 4},
            {"matrix": [3, 4], "x": 56, "y": 39, "flags": 4},
            {"matrix": [3, 5], "x": 68, "y": 39, "flags": 4},
            {"matrix": [3, 6], "x": 80, "y": 39, "flags": 4},
            {"matrix": [3, 7], "x": 92, "y": 39, "flags": 4},
            {"matrix": [3, 8], "x": 104, "y": 39, "flags": 4},
            {"matrix": [3, 9], "x": 120, "y": 39, "flags": 4},
            {"matrix": [3, 10], "x": 132, "y": 39, "flags": 4},
            {"matrix": [3, 11], "x": 132, "y": 39, "flags": 4},
            {"matrix": [3, 13], "x": 144, "y": 39, "flags": 1},
            {"matrix": [3, 14], "x": 176, "y": 39, "flags": 1},
            {"matrix": [3, 15], "x": 188, "y": 39, "flags": 4},
            {"matrix": [3, 16], "x": 200, "y": 39, "flags": 4},
            {"matrix": [3, 17], "x": 212, "y": 39, "flags": 4},
            {"matrix": [3, 18], "x": 224, "y": 39, "flags": 1},
            {"matrix": [4, 0], "x": 0, "y": 52, "flags": 1},
            {"matrix": [4, 1], "x": 26, "y": 52, "flags": 4},
            {"matrix": [4, 2], "x": 38, "y": 52, "flags": 4},
            {"matrix": [4, 3], "x": 50, "y": 52, "flags": 4},
            {"matrix": [4, 4], "x": 62, "y": 52, "flags": 4},
            {"matrix": [4, 5], "x": 74, "y": 52, "flags": 4},
            {"matrix": [4, 6], "x": 86, "y": 52, "flags": 4},
            {"matrix": [4, 7], "x": 98, "y": 52, "flags": 4},
            {"matrix": [4, 8], "x": 110, "y": 52, "flags": 4},
            {"matrix": [4, 9], "x": 122, "y": 52, "flags": 4},
            {"matrix": [4, 10], "x": 134, "y": 52, "flags": 4},
            {"matrix": [4, 11], "x": 150, "y": 52, "flags": 1},
            {"matrix": [4, 13], "x": 170, "y": 52, "flags": 1},
            {"matrix": [4, 15], "x": 188, "y": 52, "flags": 4},
            {"matrix": [4, 16], "x": 200, "y": 52, "flags": 4},
            {"matrix": [4, 17], "x": 212, "y": 52, "flags": 4},
            {"matrix": [4, 18], "x": 224, "y": 52, "flags": 1},
            {"matrix": [5, 0], "x": 0, "y": 64, "flags": 1},
            {"matrix": [5, 1], "x": 14, "y": 64, "flags": 1},
            {"matrix": [5, 2], "x": 28, "y": 64, "flags": 1},
            {"matrix": [5, 3], "x": 42, "y": 64, "flags": 4},
            {"matrix": [5, 5], "x": 74, "y": 64, "flags": 4},
            {"matrix": [5, 8], "x": 108, "y": 64, "flags": 4},
            {"matrix": [5, 9], "x": 122, "y": 64, "flags": 1},
            {"matrix": [5, 10], "x": 140, "y": 64, "flags": 1},
            {"matrix": [5, 12], "x": 158, "y": 64, "flags": 1},
            {"matrix": [5, 13], "x": 170, "y": 64, "flags": 1},
            {"matrix": [5, 14], "x": 182, "y": 64, "flags": 1},
            {"matrix": [5, 15], "x": 196, "y": 64, "flags": 4},
            {"matrix": [5, 17], "x": 212, "y": 64, "flags": 4}
        ],
        "animations": {
```

File: keyboards/qwertykeys/qk100/ansi/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+	),
+	[1] = LAYOUT(
+		KC_F14,  KC_F15,  KC_F16,  KC_F17,  KC_F18,  KC_F19,  KC_F20,  KC_F21,  KC_F22,  KC_F23,  KC_F24,  KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, NK_TOGG, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, RGB_TOG, RGB_MOD, RGB_RMOD,RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAD, RGB_VAI, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,                            KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS
+	)
+};
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/ansi/keymaps/default/keymap.c:39]: Please replace all of the tabs with spaces.

File: keyboards/qwertykeys/qk100/ansi/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+
+	[0] = LAYOUT(
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/ansi/keymaps/via/keymap.c:23]: same here.

File: keyboards/qwertykeys/qk100/ansi/rules.mk [hunk 13]
```diff
@@ -0,0 +1,5 @@
+MCU_LDSCRIPT = STM32F103x8_0x08006000_bootloader
+DFU_ARGS = -d 1688:2220 -a 0 -s 0x08006000 
+DFU_SUFFIX_ARGS = -v 1688 -p 2220
+BOOTLOADER_TYPE = stm32duino
+OPT_DEFS += -DBOOTLOADER_STM32DUINO
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/ansi/rules.mk:5]:
```suggestion
```

File: keyboards/qwertykeys/qk100/info.json [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "extrakey": true,
+        "mousekey": true,
+	    "rgblight": true,
+        "nkro": true
+    },
+    "processor": "STM32F103",
+    "board": "STM32_F103_STM32DUINO",
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/info.json:21]: This should set most of the bootloader config, minus the custom ld script and dfu suffix/args
```suggestion
    "bootloader": "stm32duino",
```

File: keyboards/qwertykeys/qk100/ld/STM32F103x8_0x08006000_bootloader.ld [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+/*
+ * STM32F103x8 memory setup for use with the STM32Duino bootloader.
+ */
+f103_flash_size = 128k;
+
+MEMORY
+{
+    /*Here we give 22K flash to bootloader*/
+    /*0x08005800 : the application bin start address*/
+    flash0  : org = 0x08006000, len = f103_flash_size - 0x6000
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/ld/STM32F103x8_0x08006000_bootloader.ld:26]: This is unusual config, and deviates from the normal stm32duino bootloader config.  Is there a specific reason for this?

File: keyboards/qwertykeys/qk100/solder/config.h [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#define DYNAMIC_KEYMAP_LAYER_COUNT 2
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/solder/config.h:20]: This should be moved to the info.json as "dynamic_keymap.layer_count"

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/qwertykeys/qk100/ansi/ansi.c [hunk 17]
```diff
@@ -0,0 +1,214 @@
...
+    {0, CS11_SW9, CS12_SW9, CS10_SW9}, /* RGB98-right */
+    {0, CS14_SW9, CS15_SW9, CS13_SW9},  /* RGB99- 0 */
+    {0, CS20_SW9, CS21_SW9, CS19_SW9}  /* RGB100- . */
+};
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+  if (record->event.pressed) {
+    switch(keycode) {
+
+        case KC_F14:
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/ansi/ansi.c:133]: These should be separate keycodes.  `QK_KB`-`QK_KB_MAX` is a range set aside explicitly for keyboard custom keycodes.

File: keyboards/qwertykeys/qk100/ansi/config.h (hunk 5 above)
Comment (drashna) [reply keyboards/qwertykeys/qk100/ansi/config.h:36]: setting the disable after timeout to 0 disables it, so there is no point in having it defined as 0.

Additionally, the rgb keycodes shouldn't be using the function keys, but should be using custom keycodes. (eg `QK_KB`-`QK_KB_MAX`)

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by owlab-git** [comment]
@drashna Thank you so much for your great advice.

File: keyboards/qwertykeys/qk100/info.json [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+    "matrix_pins": {
+        "cols": ["A7", "B0", "B1", "B2", "B10", "B11", "B12", "B13", "B14", "B15", "A8", "A9", "A10", "A15", "B4", "B5", "B8", "B9", "C13" ],
+        "rows": ["A1", "A2", "A3", "A5", "A4", "A6" ]
+    },
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "extrakey": true,
+        "mousekey": true,
+	    "rgblight": true,
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/info.json:17]:
```suggestion
        "rgblight": true,
```

File: keyboards/qwertykeys/qk100/solder/info.json
Comment (drashna) [inline keyboards/qwertykeys/qk100/solder/info.json:1]: Could you replace the tabs with spaces in this file?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/qwertykeys/qk100/ansi/config.h [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED  // turn off effects when suspended
+#    define RGB_MATRIX_KEYPRESSES           // reacts to keypresses
+#    define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#    define RGB_MATRIX_LED_PROCESS_LIMIT  4
+#    define RGB_MATRIX_LED_FLUSH_LIMIT    26
+#    define RGB_MATRIX_STARTUP_VAL        128
+#    define DRIVER_ADDR_1                 0b0110000
+#    define DRIVER_COUNT                  1
+#    define RGB_MATRIX_LED_COUNT          101 
+#endif
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/ansi/config.h:45]:
```suggestion
```

File: keyboards/qwertykeys/qk100/ansi/rules.mk [hunk 20]
```diff
@@ -0,0 +1,2 @@
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/ansi/rules.mk:2]: This is no longer needed, as it's set by default.

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by owlab-git** [comment]
Hi, Please check. Thank you for your time

**Review by drashna — APPROVED** [review]

**Comment by owlab-git** [comment]
Bump

**Comment by owlab-git** [comment]
Hello. Are there any further issues with merge?

File: keyboards/qwertykeys/qk100/ansi/chconf.h
Comment (drashna) [inline keyboards/qwertykeys/qk100/ansi/chconf.h:1]: Are these changes necessary, or just copyied over from other boards?  especially as the solder version doesn't have these.

File: keyboards/qwertykeys/qk100/ansi/halconf.h [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define HAL_USE_I2C TRUE
+
+#define HAL_USE_PWM TRUE
+
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/ansi/halconf.h:22]: Is PWM being used? 

If not: 
```suggestion
```

File: keyboards/qwertykeys/qk100/ansi/mcuconf.h [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+/*
+ * I2C driver system settings.
+ */
+#undef STM32_I2C_USE_I2C1
+#define STM32_I2C_USE_I2C1                  TRUE
+
+
+
+#undef STM32_PWM_USE_TIM1
+#define STM32_PWM_USE_TIM1 TRUE
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/ansi/mcuconf.h:30]: Same.  If PWM isn't in use:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/qwertykeys/qk100/ansi/ansi.c [hunk 23]
```diff
@@ -0,0 +1,197 @@
...
+    {0, CS38_SW6, CS39_SW6, CS37_SW6}, /* RGB97-down */
+    {0, CS11_SW9, CS12_SW9, CS10_SW9}, /* RGB98-right */
+    {0, CS14_SW9, CS15_SW9, CS13_SW9}, /* RGB99- 0 */
+    {0, CS20_SW9, CS21_SW9, CS19_SW9}  /* RGB100- . */
+};
+
+bool process_record_kb(uint16_t keycode, keyrecord_t* record) {
+    if (record->event.pressed) {
+        switch (keycode) {
+            case QK_KB_0:
```
Comment (drashna) [inline keyboards/qwertykeys/qk100/ansi/ansi.c:127]: For reference, here is how the normal keycodes function: 
https://github.com/qmk/qmk_firmware/blob/master/quantum/process_keycode/process_rgb.c

If you're fine with this, awesome. But just in case you want to have them handled like the normal RGB keycodes, you can do so.

**Review by drashna — APPROVED** [review]

File: keyboards/qwertykeys/qk100/ansi/ansi.c (hunk 23 above)
Comment (owlab-git) [reply keyboards/qwertykeys/qk100/ansi/ansi.c:127]: Yes, we do  want to have them handled like the normal RGB keycodes. Thank you very much for your suggestion anyway.

**Comment by owlab-git** [comment]
Please Check for merge.

**Comment by owlab-git** [comment]
Bump

**Comment by owlab-git** [comment]
Please check, thanks.

**Review by waffle87 — APPROVED** [review]
